QuizApp/QuizApp/Main/AppDependencies.swift

```diff
@@ -149,6 +154,13 @@ class AppDependencies {
             }
             .scope(.unique)
 
+        container
+            .register { (_, args) -> LeaderboardViewModel in
+                let id: Int = args.get()
```

Comment:
Can you try
`LeaderboardViewModel(id: args.get(), useCase: container.resolve(), router: container.resolve())`